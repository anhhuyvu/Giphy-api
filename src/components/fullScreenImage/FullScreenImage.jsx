import React, { useEffect } from 'react';
import './FullScreenImage.scss';

const FullScreenImage = ({ isOpen, url, toggle }) => {
  useEffect(() => {
    const handleOnClick = (event) => {
      const container = document.getElementById('fullscreen-container');

      if (container && container.isEqualNode(event.target)) {
        toggle();
      }
    };

    document.addEventListener('mousedown', handleOnClick);

    return () => {
      document.removeEventListener('mousedown', handleOnClick);
    };
  }, [toggle]);

  return (
    <div className={`fullscreen-container ${isOpen ? 'show' : ''}`} id="fullscreen-container">
      <img alt={url} src={url} />
    </div>
  );
};

export default FullScreenImage;
