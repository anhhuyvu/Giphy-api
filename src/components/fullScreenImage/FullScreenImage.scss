.fullscreen-container {
  top: 0;
  left: 0;
  opacity: 0;
  z-index: -1;
  width: 100vw;
  height: 100vh;
  padding: 1rem;
  position: fixed;
  transition: all ease 0.2s;
  background-color: rgba(40, 40, 40, 0.9);

  img {
    pointer-events: none;
    width: calc(100vw - 2rem);
    height: calc(100vh - 2rem);
    object-fit: contain;
  }

  &.show {
    opacity: 1;
    z-index: 2;
  }
}
