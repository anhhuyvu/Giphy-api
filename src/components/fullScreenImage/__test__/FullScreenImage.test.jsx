import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import FullScreenImage from '../FullScreenImage';
describe('FullScreenImage component', () => {
  it('should render snapshot correctly', () => {
    const tree = renderer.create(<FullScreenImage url={'url'} isOpen={true} toggle={'toggle'} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render overlay image', () => {
    const wrapper = render(<FullScreenImage url={'url'} isOpen={true} toggle={'toggle'} />);

    expect(wrapper.getByRole('img')).toBeDefined();
  });
});
