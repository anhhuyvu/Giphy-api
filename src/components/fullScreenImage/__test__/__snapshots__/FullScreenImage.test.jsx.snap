// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullScreenImage component should render snapshot correctly 1`] = `
<div
  className="fullscreen-container show"
  id="fullscreen-container"
>
  <img
    alt="url"
    src="url"
  />
</div>
`;
